include/sensors/core/mmr920C04_registers.hpp

```diff
@@ -0,0 +1,313 @@
+    static constexpr bool readable = true;
+    static constexpr bool writable = false;
+    static constexpr uint32_t value_mask = (1 << 8) - 1;
+
+    // Pascals per 1 cmH20
+    static constexpr float CMH20_TO_PASCALS = 98.0665;
+    static constexpr float PA_PER_COUNT =
+        1e-5 * CMH20_TO_PASCALS;  // 1.0e-5cmH2O/count * 98.0665Pa/cmH2O
+
+    uint32_t C7 : 1 = 1;
+    uint32_t C6 : 1 = 1;
+    uint32_t C5 : 1 = 0;
+    uint32_t C4 : 1 = 0;
+    uint32_t C3 : 1 = 0;
+    uint32_t C2 : 1 = 1;
+    uint32_t C1 : 1 = 0;
+    uint32_t C0 : 1 = 0;
+    uint32_t reading : 24 = 0;
+
+    [[nodiscard]] static auto to_pressure(uint32_t reg) -> sq14_15 {
+        // Sign extend pressure result
+        if (reg & 0x00800000)
+            reg |= 0xFF000000;
+        else
+            reg &= 0x007FFFFF;
+
+        float pressure = static_cast<float>(reg) * PA_PER_COUNT;
+        return convert_to_fixed_point(pressure, 15);
+    }
+};
+
+struct __attribute__((packed, __may_alias__)) Temperature {
+    static constexpr Registers address = Registers::TEMPERATURE_READ;
+    static constexpr bool readable = true;
+    static constexpr bool writable = false;
+    static constexpr uint32_t value_mask = (1 << 24) - 1;
+
+    static constexpr uint32_t MAX_SIZE = (2 << 7);
+
+    static constexpr float CONVERT_TO_CELSIUS = 0.0078125;
+
+    uint32_t C7 : 1 = 1;
+    uint32_t C6 : 1 = 1;
+    uint32_t C5 : 1 = 0;
+    uint32_t C4 : 1 = 0;
+    uint32_t C3 : 1 = 0;
+    uint32_t C2 : 1 = 0;
+    uint32_t C1 : 1 = 1;
+    uint32_t C0 : 1 = 0;
+    uint32_t reading : 24 = 0;
+
+    [[nodiscard]] static auto to_temperature(uint32_t reg) -> sq14_15 {
+        float temperature =
+            CONVERT_TO_CELSIUS * (static_cast<float>(reg) / MAX_SIZE);
+        return convert_to_fixed_point(temperature, 15);
+    }
+};
+
+struct __attribute__((packed, __may_alias__)) Status {
+    static constexpr Registers address = Registers::STATUS;
+    static constexpr bool readable = true;
+    static constexpr bool writable = false;
+    static constexpr uint8_t value_mask = (1 << 8) - 1;
+
+    uint32_t C7 : 1 = 1;
+    uint32_t C6 : 1 = 0;
+    uint32_t C5 : 1 = 0;
+    uint32_t C4 : 1 = 0;
+    uint32_t C3 : 1 = 0;
+    uint32_t C2 : 1 = 0;
+    uint32_t C1 : 1 = 0;
+    uint32_t C0 : 1 = 0;
+    uint32_t reading : 8 = 0;
+
+    [[nodiscard]] static auto to_status(uint8_t reg) -> SensorStatus {
+        switch (static_cast<SensorStatus>(reg)) {
+            case SensorStatus::IDLE:
+                return SensorStatus::IDLE;
+            case SensorStatus::ACTIVE:
+                return SensorStatus::ACTIVE;
+            case SensorStatus::SHUTDOWN:
+                return SensorStatus::SHUTDOWN;
+            default:
+                return SensorStatus::UNKNOWN;
+        }
+    }
+};
+
+struct MMR920C04RegisterMap {
+    Reset reset = {};
+    Idle idle = {};
+    MeasureMode1 measure_mode_1 = {};
```

Comment:
What are the differences between these measuring modes?

Reply:
It affects the low pass filter option of pressure readings